packages/react-components/react-utilities/src/hooks/useMergedEventCallbacks.ts

```diff
@@ -1,29 +1,13 @@
 import { useEventCallback } from './useEventCallback';
+import { mergeCallbacks } from '../utils/mergeCallbacks';
 
 /**
  * @internal
- * Combine two event callbacks into a single callback function that calls each one in order.
- *
- * This is useful to add an event listener to an existing element without overwriting the current listener, if any.
- *
- * For example:
- * ```ts
- * state.slot.onChange = useMergedCallbacks(state.slot.onChange, ev => {
- *   // Handle onChange
- * });
- * ```
- *
- * @param callback1 - The first callback to be called
- * @param callback2 - The second callback to be called
- *
- * @returns An event callback that calls the callbacks in order, and is stable between renders
+ * @deprecated useEventCallback(mergeCallbacks(callback1, callback2))
```

Comment:
This has been `@internal` function, cannot we just remove it without deprecation?

Reply:
> LGTM.
> 
> I still don't see the point of having any of those functions to TBH, I don't see much win on using this instead of simply calling the users callback inside our own internal handlers 😅.
> 
> Would be nice to have at least a comment or something on the advantages of this approach, as IMO it is less readable.

The primary benefit is to help with capturing the existing callback safely, which would require saving the existing callback to a local variable and calling the local variable. 

```ts
// To be correct, you must save the existing callback to a local variable and call it.
const originalOnChange = state.slot.onChange;
state.slot.onChange = (ev, data) => {
  originalOnChange?.(ev, data);
  // Handle onChange
};

// ❌ This would cause infinite recursion.
state.slot.onChange = (ev, data) => {
  state.slot.onChange?.(ev, data);
  // Handle onChange
};
```

`mergeCallbacks` turns merging callbacks into a single statement, which is nice especially if overriding a bunch of callbacks.


```ts
state.slot.onChange = mergeCallbacks(state.slot.onChange, (ev, data) => {
  // Handle onChange
});
```

Reply:
Ok I'll remove useMergedEventCallbacks instead of deprecating it.